### PR TITLE
Feature/role edit form

### PR DESCRIPTION
**Pull request by NicholasIbarra**
Implemented Roles Management UI
- Create role with ability to copy permissions from another role
- Edit role name \ description
- Edit permissions 
- Updated controllers to support workflows

Permissions UI allows to select all permissions by group or un-select all. 
Permissions are grouped by resource and show a badge of how many permissions are selected. Shows "all" if all are selected

## Create Role Form
image

## Edit Role Form
image

## Edit Permissions Form
image

### CODE REVIEW

File: Moda.Web/src/moda.web.reactclient/src/app/settings/roles/[id]/page.tsx [hunk 1]
```diff
@@ -1,18 +1,69 @@
...
 
   return (
     <>
       <PageTitle
-        title={role?.name ?? `Test ${params.id}`}
+        title={roleData.data?.name}
         subtitle="Role Details"
       />
+      <Card
+        style={{ width: '100%', height: '100%' }}
```
[the remark this answers is not in the document]
Comment (NicholasIbarra) [reply Moda.Web/src/moda.web.reactclient/src/app/settings/roles/[id]/page.tsx]: I noticed there is a scroll, but that was from the head. For consistency, I removed the height 100%

File: Moda.Web/src/moda.web.reactclient/src/app/settings/roles/[id]/components/detail.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          onFinishFailed={onFinishFailed}
+          autoComplete="off"
+        >
+          <Form.Item
+            label="Name"
+            name="name"
+            initialValue={role?.name}
+            rules={[{ required: true, message: 'Please input a name' }]}
+          >
+            <Input />
```
[the remark this answers is not in the document]
Comment (NicholasIbarra) [reply Moda.Web/src/moda.web.reactclient/src/app/settings/roles/[id]/components/detail.tsx]: I've added to name, but not description. The database has description as an nvarchar(max).

File: Moda.Web/src/moda.web.reactclient/src/app/settings/roles/[id]/components/detail.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        <Form
+          labelCol={{ span: 8 }}
+          wrapperCol={{ span: 16 }}
+          style={{ maxWidth: 600 }}
+          onFinish={onFinish}
+          onFinishFailed={onFinishFailed}
+          autoComplete="off"
+        >
+          <Form.Item
+            label="Name"
```
[the remark this answers is not in the document]
Comment (NicholasIbarra) [reply Moda.Web/src/moda.web.reactclient/src/app/settings/roles/[id]/components/detail.tsx:77]: I limited the form now to 256, do you think that suffices or should I bind server errors?

[the remark this answers is not in the document]
Comment (destacey) [reply Moda.Web/src/moda.web.reactclient/src/app/settings/roles/[id]/components/detail.tsx:77]: Looks good.  We can bind the server errors later.

**Review by destacey — APPROVED** [review]

**Pull request merged**